### PR TITLE
feat(app): add network flow

### LINKED ISSUE #15: Create NetworkForm component
## Figma Reference
https://www.figma.com/file/uEckQY4EaQ5rXwyISg6d1g/Fuel-Wallet?node-id=290%3A16258

## Props
- `network?: Network`

## Form fields
- `name: String`
- `url: String`

### LINKED ISSUE #16: Create Networks screen
## Tasks

- [ ] Just use a `Stack` component to render the list
- [ ] Add a `Button` component as `Add new network` link

## Blocked by
- #14 
- #15

### LINKED ISSUE #17: Create NetworkEdit screen
## Observations
- Will use the `NetworkForm` component to render an edit screen

## Blocked by
- #15

### LINKED ISSUE #18: Create NetworkCreate screen
## Observations
- Will use the `NetworkForm` component to render a create screen

## Blocked by
- #15

### LINKED ISSUE #44: Create NetworkDropdown component
## Figma Reference
https://www.figma.com/file/uEckQY4EaQ5rXwyISg6d1g/Fuel-Wallet?node-id=77%3A7512

## Props
- `network: Network`

## Blocked by
- #14

### PR DESCRIPTION
**Pull request by pedronauck**
✨ Fix #44 
✨ Fix #17
✨ Fix #18
✨ Fix #16
✨ Fix #15

### CODE REVIEW

**Review by luizstacio — CHANGES_REQUESTED** [review]
- Try to split this PR into more small scopes
- When accessing the storybook some components are not working 
Screen Shot 2022-09-27 at 11 08 32 AM
- When finishing creating a network it could go back home with the new network already selected.

File: packages/app/package.json [hunk 1]
```diff
@@ -40,13 +42,16 @@
...
+    "query-string": "^7.1.1",
     "react": "^18.2.0",
     "react-content-loader": "^6.2.0",
     "react-dom": "^18.2.0",
     "react-google-recaptcha": "^2.1.0",
     "react-helmet": "^6.1.0",
     "react-hook-form": "^7.36.0",
     "react-router-dom": "^6.4.0",
+    "short-unique-id": "^4.4.4",
+    "storybook-addon-react-router-v6": "^0.2.1",
```
[the remark this answers is not in the document]
Comment (pedronauck) [reply packages/app/package.json]: ✅

File: packages/app/cypress/e2e/FaucetDialog.cy.ts [hunk 2]
```diff
@@ -7,20 +7,11 @@ describe('FaucetDialog', () => {
   });
 
   it('should be able to faucet and get redirected to home again with balances', () => {
-    /** Checks empty balance */
```
[the remark this answers is not in the document]
Comment (pedronauck) [reply packages/app/cypress/e2e/FaucetDialog.cy.ts]: ✅

File: packages/app/src/main.tsx [hunk 3]
```diff
@@ -4,6 +4,14 @@ import { BrowserRouter } from "react-router-dom";
 
 import { App } from "./App";
 
+if (import.meta.env.DEBUG === "test") {
```
[the remark this answers is not in the document]
Comment (pedronauck) [reply packages/app/src/main.tsx]: ✅

File: packages/app/src/systems/Core/components/Layout/Layout.tsx [hunk 4]
```diff
@@ -45,22 +48,22 @@ type LayoutComponent = FC<LayoutProps> & {
...
       <Helmet>
         <title>{titleText}</title>
       </Helmet>
       <Flex as="main" css={styles.root({ isPublic })}>
-        {isPublic ? (
-          <>{children}</>
-        ) : (
-          <Flex css={styles.wrapper}>{children}</Flex>
-        )}
+        {isPublic ? <>{children}</> : <Flex css={styles.wrapper}>{children}</Flex>}
```
Comment (luizstacio) [inline packages/app/src/systems/Core/components/Layout/Layout.tsx:62]: Can we keep the rule for didn't change these cases? It's more readable with the `()`

File: packages/app/src/systems/Core/components/Layout/Layout.tsx [hunk 5]
```diff
@@ -45,22 +48,22 @@ type LayoutComponent = FC<LayoutProps> & {
...
       </Flex>
+      {import.meta.env.NODE_ENV === 'test' && (
```
Comment (luizstacio) [inline packages/app/src/systems/Core/components/Layout/Layout.tsx:64]: This is just for cypress checking?

File: packages/app/src/systems/SignUp/components/MnemonicRead/MnemonicRead.tsx [hunk 6]
```diff
@@ -38,18 +35,10 @@ export function MnemonicRead({ words, onCancel, onNext }: MnemonicReadProps) {
...
       </Form.Control>
       <Flex gap="$4">
-        <Button
-          size="sm"
-          color="gray"
-          variant="ghost"
-          css={{ width: 130 }}
-          onPress={onCancel}
-        >
+        <Button size="sm" color="gray" variant="ghost" css={{ width: 130 }} onPress={onCancel}>
```
Comment (luizstacio) [inline packages/app/src/systems/SignUp/components/MnemonicRead/MnemonicRead.tsx:41]: Also, the old rule for formatting in multiple lines is better here.

File: packages/app/src/systems/SignUp/components/MnemonicWrite/MnemonicWrite.test.tsx [hunk 7]
```diff
@@ -1,21 +1,19 @@
...
+import { getPhraseFromValue } from '~/systems/Core';
 
 const onFilledHandler = jest.fn();
 const onNextHandler = jest.fn();
 const onCancelHandler = jest.fn();
 
-const MNEMONIC = getPhraseFromValue(
-  FuelMnemonic.generate(MNEMONIC_SIZE)
-) as string;
+const MNEMONIC = getPhraseFromValue(FuelMnemonic.generate(MNEMONIC_SIZE)) as string;
```
Comment (luizstacio) [inline packages/app/src/systems/SignUp/components/MnemonicWrite/MnemonicWrite.test.tsx:13]: Also here the old rule for formatting was better, maybe the column number should be smaller

File: packages/app/src/systems/SignUp/pages/CreateWallet/CreateWallet.tsx [hunk 8]
```diff
@@ -1,46 +1,41 @@
...
 
   return (
     <Layout title="Create Wallet" isPublic>
-      {state.matches("showingMnemonic") && (
+      {state.matches('showingMnemonic') && (
         <MnemonicRead
           words={context.data?.mnemonic}
           onNext={handlers.next}
-          onCancel={() => navigate("/sign-up")}
+          onCancel={() => navigate('/sign-up')}
```
Comment (luizstacio) [inline packages/app/src/systems/SignUp/pages/CreateWallet/CreateWallet.tsx:19]: Change this navigates to use the Pages.<route>()

File: packages/app/src/systems/SignUp/pages/RecoverWallet/RecoverWallet.tsx [hunk 9]
```diff
@@ -21,43 +16,43 @@ export function RecoverWallet() {
...
       </Alert>
-      {state.matches("waitingMnemonic") && (
+      {state.matches('waitingMnemonic') && (
         <MnemonicWrite
           error={handlers.checkMnemonicError()}
           canProceed={context.isConfirmed}
           onFilled={handlers.confirmMnemonic}
           onNext={handlers.next}
-          onCancel={() => navigate("/sign-up")}
+          onCancel={() => navigate('/sign-up')}
```
Comment (luizstacio) [inline packages/app/src/systems/SignUp/pages/RecoverWallet/RecoverWallet.tsx:29]: Also here: Pages.<route>() look for other places in this code

**Review by luizstacio — CHANGES_REQUESTED** [review]

**Comment by luizstacio** [comment]
Move the loader spinner to the other side of the network selector to prevent the selector from changing the page on after the loading state.
Screen Shot 2022-09-29 at 10 54 13 PM

File: .prettierrc [hunk 10]
```diff
@@ -1,7 +1,7 @@
 {
   "overrides": [
     {
-      "files": ["*.js", "*.ts"],
+      "files": ["*.js", "*.ts", "*.jsx", "*.tsx"],
       "options": {
         "printWidth": 100,
```
Comment (luizstacio) [inline .prettierrc:6]: "printWidth": 80

**Review by luizstacio — APPROVED** [review]

**Pull request merged**